docs/getting-started/domain.md

```diff
@@ -0,0 +1,37 @@
+# Domain Model
+
+## Step
+
+A step is a named action that needs to be performed, usually a single shell command.  If the command finishes with a non-zero exit code, the step is considered a failure.
+
+## Metadata
+
+Metadata is a structured key/value storage of relevant information about an instance of a Job.  This is automatically populated with basic information like Git SHA1 and start/stop time.  It can be updated throughout the job.
+
+## Job
+
+A job consists of multiple steps, each of which will be run in series.  If any step fails, then the entire job is considered failed and subsequent steps will be skipped (unless configured otherwise).
+
+Jobs share four pieces of context between steps:
+
+ - Source Code - Checked out to a given revision at the start of the job
+ - Filesystem - Any changes made by a step to the filesystem will be visible to subsequent steps
+ - Container - Any processes started or stopped by a step will be visible to subsequent steps
+ - Metadata - Structured key/value storage of relevant information
+
+Jobs are started by changes in source code, as a trigger from an upstream job, or by human decision.
```

Comment:
Various events can trigger jobs to start, such as changing the source code, completing an upstream job, or manually clicking `Start` in the UI.